core/src/main/java/org/apache/calcite/tools/Programs.java

```diff
@@ -266,15 +266,14 @@ public static Program standard() {
   public static Program standard(RelMetadataProvider metadataProvider) {
     final Program program1 =
         (planner, rel, requiredOutputTraits, materializations, lattices) -> {
-          planner.setRoot(rel);
```

Comment:
Setting root before addMaterialization will make MaterializedViewRule always fail to match